src/pages/search.vue

```diff
@@ -35,14 +28,31 @@ import {
 import { SET_FILTER_IS_VISIBLE } from '~/constants/mutation-types'
 import { queryStringToSearchType } from '~/utils/search-query-transform'
 import local from '~/utils/local'
-import { screenWidth } from '~/utils/get-browser-info'
 import { ALL_MEDIA, IMAGE, VIDEO } from '~/constants/media'
 import { mapActions, mapGetters, mapMutations, mapState } from 'vuex'
 import { MEDIA, SEARCH } from '~/constants/store-modules'
 import debounce from 'lodash.debounce'
+import SearchGridFilter from '~/components/Filters/SearchGridFilter.vue'
+import { useMediaQuery } from '~/composables/use-media-query.js'
+import AppModal from '~/components/AppModal'
+
+const MIN_SCREEN_WIDTH_FILTER_VISIBLE_BY_DEFAULT = 768
 
 const BrowsePage = {
   name: 'browse-page',
+  components: {
+    SearchGridFilter,
+  },
+  setup() {
+    const defaultWindow = typeof window !== 'undefined' ? window : undefined
+    const isLargeScreen = useMediaQuery(
+      `(min-width: ${MIN_SCREEN_WIDTH_FILTER_VISIBLE_BY_DEFAULT}px)`,
+      defaultWindow
+    )
+    return {
+      isLargeScreen,
+    }
```

Comment:
Mixing `setup()` with the older options API is not very nice. Moving `computed` inside `setup` would be better.

Reply:
I agree and also would prefer not to mix the two Vue API, but to change it completely will require to move methods and watchers too, and this page component depends heavily on the global state using Vuex, whose current functioning as we know it will probably change. I'm honestly wondering if it's worth changing the rest to the composition API if there is no a clear conflict to resolve and given the [`useStore` situation in Nuxt3](https://github.com/nuxt/framework/issues/899).